File: src/commands/ScoreboardCommand.ts
import { ScoreboardManager } from "../scoreboard/ScoreboardManager";
import { CommandContext } from "../types/hytale";

export class ScoreboardCommand {
  private readonly manager: ScoreboardManager;

  constructor(manager: ScoreboardManager) {
    this.manager = manager;
  }

  handle(context: CommandContext): void {
    const [subcommand] = context.args;

    if (!subcommand) {
      context.reply("Utilisation: /scoreboard <on|off|test>");
      return;
    }

    switch (subcommand.toLowerCase()) {
      case "on":
        this.manager.setEnabled(context.player, true);
        this.manager.setTestMode(context.player, false);
        context.reply("Scoreboard activé.");
        return;
      case "off":
        this.manager.setEnabled(context.player, false);
        this.manager.setTestMode(context.player, false);
        context.reply("Scoreboard désactivé.");
        return;
      case "test":
        this.manager.setEnabled(context.player, true);
        this.manager.setTestMode(context.player, true);
        context.reply("Scoreboard en mode test.");
        return;
      default:
        context.reply("Utilisation: /scoreboard <on|off|test>");
    }
  }
}

File: src/main/java/com/hytale/scoreboard/commands/ScoreboardCommand.java
package com.hytale.scoreboard.commands;

import com.hytale.scoreboard.scoreboard.ScoreboardManager;
import com.hytale.scoreboard.types.CommandContext;

public class ScoreboardCommand {
  private final ScoreboardManager manager;

  public ScoreboardCommand(ScoreboardManager manager) {
    this.manager = manager;
  }

  public void handle(CommandContext context) {
    String subcommand = context.getArgs().isEmpty() ? null : context.getArgs().get(0);

    if (subcommand == null) {
      context.reply("Utilisation: /scoreboard <on|off|test>");
      return;
    }

    switch (subcommand.toLowerCase()) {
      case "on":
        manager.setEnabled(context.getPlayer(), true);
        manager.setTestMode(context.getPlayer(), false);
        context.reply("Scoreboard activé.");
        return;
      case "off":
        manager.setEnabled(context.getPlayer(), false);
        manager.setTestMode(context.getPlayer(), false);
        context.reply("Scoreboard désactivé.");
        return;
      case "test":
        manager.setEnabled(context.getPlayer(), true);
        manager.setTestMode(context.getPlayer(), true);
        context.reply("Scoreboard en mode test.");
        return;
      default:
        context.reply("Utilisation: /scoreboard <on|off|test>");
    }
  }
}
